core/services/relay/evm/chain_writer.go

```diff
@@ -156,7 +169,38 @@ func (w *chainWriter) GetTransactionStatus(ctx context.Context, transactionID uu
 }
 
 func (w *chainWriter) GetFeeComponents(ctx context.Context) (*commontypes.ChainFeeComponents, error) {
-	return nil, fmt.Errorf("not implemented")
+	if w.ge.estimator == nil {
+		return nil, fmt.Errorf("not implemented")
+	}
+
+	gasPriceWei, _, err := w.ge.estimator.GetFee(ctx, nil, 0, assets.NewWei(w.ge.maxGasPrice))
+	if err != nil {
+		return nil, err
+	}
+	// Use legacy if no dynamic is available.
+	gasPrice := gasPriceWei.Legacy.ToInt()
```

Comment:
I know this code here is copied from somewhere else, but I think it'd be helpful to describe what's going on here in the comments. In particular, what's the difference between the DynamicFeeCap and the Legacy Gas Price? What's the situation in which we fall back to legacy, ie why would the DynamicFeeCap be nil?

Reply:
Added comment explaining this above the method signature